app/views/handlers/list_action.py

```diff
@@ -47,8 +49,33 @@ def handle_post(self):
             self.parent_location.block_id
         )
         self.questionnaire_store_updater.remove_answers(answer_ids_to_remove)
+        self.evaluate_sections()
         self.questionnaire_store_updater.save()
 
+    def evaluate_sections(self):
```

Comment:
This is not just evaluating, it is updating so the method name should reflect that.